### DISCUSSION TITLE
PIP-20: Basic Authentication for gRPC APIs

### DISCUSSION THREAD

**Discussion by ja7ad** [pactus-project/PIPs#87] [PIPs]
PIP-20: [Basic Authentication for gRPC APIs ](https://pips.pactus.org/PIPs/pip-20)

**Comment by ja7ad** [comment]
@b00f @kehiy @amirvalhalla Any technical note for this proposal? If this proposal is optimal we can accept it for implementation for version 1.1.0.

**Reply by b00f** [reply]
Let's have a DEV meeting for this PIP.

**Reply by ja7ad** [reply]
> Let's have a DEV meeting for this PIP.

Ok

**Reply by b00f** [reply]
For the fast verification we need to make sure the password either hash in client side or cached in server.

**Reply by ja7ad** [reply]
> For the fast verification we need to make sure the password either hash in client side or cached in server. 

Authentication is per request not have topically

**Reply by b00f** [reply]
Let's check the implementation, this should be done. Otherwise gRPC calls will be slow

**Reply by ja7ad** [reply]
> Let's check the implementation, this should be done. Otherwise gRPC calls will be slow
> 

We can test by using grpc benchmark tools

**Reply by amirvalhalla** [reply]
@Ja7ad 
I think everything is clear.
thanks

**Comment by ja7ad** [comment]
Implemented in: https://github.com/pactus-project/pactus/pull/1162

**Comment by b00f** [comment]
Cool, gRPC is much secure now.
